src/api/users/src/routes/users.js

```diff
@@ -140,6 +142,43 @@ router.put(
     const { id } = req.params;
     const { body } = req;
 
+    try {
+      const userRef = db.doc(id);
+      const doc = await userRef.get();
+
+      if (!doc.exists) {
+        next(createError(404, `user ${id} not found.`));
+      } else {
+        const user = new User(body);
+        // no user can be created as an admin by default
+        user.isAdmin = false;
+        // NOTE: doc().update() doesn't use the converter, we have to make a plain object.
+        await db.doc(id).update(user.toJSON());
+        res.status(200).json({ msg: `Updated user ${id}` });
+      }
+    } catch (err) {
+      next(err);
+    }
+  }
+);
+
+// put (update) a user with a supplied id, validated by the schema
+// rejected if a user could not be found, otherwise user updated
+// this route is only accessible by administrators
+// it allows the modification of the isAdmin property
+router.put(
+  '/:id/admin',
+  isAuthenticated(),
+  validateId(),
+  validateUser(),
+  validateEmailHash(),
+  isAuthorized((req, user) => {
```

Comment:
Arrow functions with body statements that immediately return a value should be rewritten as inline expressions:

```js
isAuthorized((req, user) => user.roles.includes('admin')),
```

Reply:
Done